scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/SortModifiers.scala

```diff
@@ -35,31 +35,48 @@ object SortModifiers extends Rewrite {
       case t: Defn.Trait => sortMods(t.mods)
       case p: Term.Param =>
         sortMods(
-          p.mods.filterNot(m => m.is[Mod.ValParam] || m.is[Mod.VarParam]))
+          p.mods.filterNot(m => m.is[Mod.ValParam] || m.is[Mod.VarParam]),
+          code = code)
     }
     patchesOfPatches.flatten
   }
 
   private def sortMods(
-      oldMods: Seq[Mod]
+      oldMods: Seq[Mod],
+      code: Tree = null
   )(implicit order: Vector[ModKey]): Seq[Patch] = {
     if (oldMods.isEmpty) Nil
     else {
-      val sortedMods: Seq[Mod] = oldMods.sortWith(orderModsBy(order))
-      sortedMods.zip(oldMods).flatMap {
+      val sanitized = oldMods.filterNot(isHiddenImplicit)
+      val sortedMods: Seq[Mod] = sanitized.sortWith(orderModsBy(order))
+
+      sortedMods.zip(sanitized).flatMap {
         case (next, old) =>
-          if (old.tokens.isEmpty) {
-            //required for cases like: def foo(implicit x: Int)
-            Nil
-          } else {
-            val removeOld = old.tokens.map(t => TokenPatch.Remove(t))
-            val addNext = TokenPatch.AddRight(old.tokens.head, next.syntax)
-            removeOld :+ addNext
-          }
+          val removeOld = old.tokens.map(t => TokenPatch.Remove(t))
+          val addNext = TokenPatch.AddRight(old.tokens.head, next.syntax)
+          removeOld :+ addNext
       }
     }
   }
 
+  /**
+    * In cases like:
+    * {{{
+    *   class X(
+    *     implicit
+    *     private[this] val i1: Int,
+    *     private[this] var i2: String
+    * )
+    * }}}
+    *
+    * `val i1`, and `var i2` have a ``Mod.Implicit`` with empty tokens.
+    * Therefore we want to completely ignore this "mod" whenever we sort,
+    * and apply patches
+    */
+  private def isHiddenImplicit(m: Mod): Boolean = {
+    m.tokens.isEmpty && m.is[Mod.Implicit]
```

Comment:
I think this will return true for `def foo(implicit x: Int)` due to a bug, can you double check?

Reply:
@olafurpg will do.

Reply:
@olafurpg indeed it does return `true`, but luckily no rewrite is necessary in this case, so the final result is OK. Added an explicit test case for this:
```
<<< 1148 — only implicit param
def foo(implicit x: Int)
>>>
def foo(implicit x: Int)
```